conda_build/build.py

```diff
@@ -674,10 +674,13 @@ def bundle_conda(output, metadata, env, **kw):
         keep_files = set(utils.expand_globs(files, metadata.config.build_prefix))
         pfx_files = set(utils.prefix_files(metadata.config.build_prefix))
         initial_files = set(item for item in (pfx_files - keep_files)
-                            if not any(keep_file.startswith(item) for keep_file in keep_files))
+                            if not any(keep_file == item for keep_file in keep_files))
 
     files = post_process_files(metadata, initial_files)
 
+    assert 'bin/conda' not in files and 'Scripts/conda.exe' not in files, ("Bug in conda-build has "
+        "included conda binary in package.  Please report this on the conda-build issue tracker.")
+
```

Comment:
heh.  Asleep at the wheel again.  Thanks.  I'll try to change the check appropriately.